File: backend/app/schemas/me.py
from typing import Optional, List, Dict, Any
from pydantic import BaseModel

class UserProfileRead(BaseModel):
    id: str
    clerkUserId: str
    email: str
    name: str
    role: str
    department: Optional[str] = None
    isActive: bool = True

class WorkspaceRead(BaseModel):
    id: str
    name: str
    slug: str
    type: str
    region: str
    scale: str
    isDemo: bool

class OnboardingProgress(BaseModel):
    completed: bool
    status: str
    nextStep: Optional[str] = None

class AvatarPreferencesDTO(BaseModel):
    enabled: bool = True
    reducedMotion: bool = False
    companionHintsEnabled: bool = True
    soundEnabled: bool = False
    avatarVariant: str = "TACTILE_SPATIAL_MINIMAL"

class BootstrapResponse(BaseModel):
    user: UserProfileRead
    workspace: Optional[WorkspaceRead] = None
    role: str
    permissions: List[str]
    onboarding: OnboardingProgress
    unreadNotifications: int
    dataFreshness: str = "FRESH"
    destination: str = "/app/overview"
    avatar: AvatarPreferencesDTO

File: frontend/lib/api/endpoints/me.ts
import { authFetch } from '../auth-fetch';

export interface BootstrapData {
  user: {
    id: string;
    clerkUserId: string;
    email: string;
    name: string;
    role: string;
    department?: string;
    isActive: boolean;
  };
  workspace?: {
    id: string;
    name: string;
    slug: string;
    type: string;
    region: string;
    scale: string;
    isDemo: boolean;
  };
  role: string;
  permissions: string[];
  onboarding: {
    completed: boolean;
    status: string;
    nextStep?: string | null;
  };
  unreadNotifications: number;
  dataFreshness: string;
  destination: string;
  avatar: {
    enabled: boolean;
    reducedMotion: boolean;
    companionHintsEnabled: boolean;
    soundEnabled: boolean;
    avatarVariant: string;
  };
}

export const meApi = {
  getBootstrap: () => authFetch<BootstrapData>('/api/v1/me/bootstrap'),
  getAvatar: () => authFetch('/api/v1/me/avatar'),
  updateAvatar: (data: any) => authFetch('/api/v1/me/avatar', { method: 'PATCH', body: JSON.stringify(data) }),
};
